data/transactions/logic/eval.go

```diff
@@ -1150,6 +1154,19 @@ func opGlobal(cx *evalContext) {
 	cx.nextpc = cx.pc + 2
 }
 
+// Msg is data meant to be signed and then verified with the
+// ed25519verify opcode.
+type Msg struct {
+	_struct     struct{}      `codec:",omitempty,omitemptyarray"`
+	ProgramHash crypto.Digest `codec:"p"`
```

Comment:
I don't like putting the program hash in here because then a signed fact can have only one consumer. An oracle can't just say "thing is true!" it has to say "thing is true, Bob" and possibly repeat that for many recipients.

Reply:
Okay, offline discussion on this is that an oracle can set up an explicit escrow account _E_ and then produce statements of the form `Sign_oracle(H(E) || stmt)`. _E_ allows group transactions in TEAL that can read off this statement. (Optionally the group constraint can require an extra fee to be paid to the oracle to read off the statement.)

Reply:
(The other outcome of the discussion is we'll maybe support either byte blob concatenation or substring slicing on the blobs, but that will be after the next release.)